src/appimagetool_fetch_runtime.c

```diff
@@ -0,0 +1,155 @@
+// need to define this to enable asprintf
+#define _GNU_SOURCE
+#include <stdbool.h>
+#include <stdio.h>
+
+#include <curl/curl.h>
+#include <malloc.h>
+#include <string.h>
+#include <errno.h>
+
+#include "appimagetool_fetch_runtime.h"
+
+bool fetch_runtime(char* arch, size_t* size, char** buffer, bool verbose) {
+    // not the cleanest approach to globally init curl here, but this method shouldn't be called more than once anyway
+    curl_global_init(CURL_GLOBAL_ALL);
+
+    char* url = NULL;
+    int url_size = asprintf(&url, "https://github.com/AppImage/type2-runtime/releases/download/continuous/runtime-%s", arch);
```

Comment:
Need to use `free` on the memory allocated by `asprintf`?

Reply:
True. I'll use a stack buffer. The URL is pretty short, after all.